examples/quantum_die.py

```diff
@@ -1,79 +1,45 @@
-#
-#    Licensed under the Apache License, Version 2.0 (the "License");
-#    you may not use this file except in compliance with the License.
-#    You may obtain a copy of the License at
-#
-#        http://www.apache.org/licenses/LICENSE-2.0
-#
-#    Unless required by applicable law or agreed to in writing, software
-#    distributed under the License is distributed on an "AS IS" BASIS,
-#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#    See the License for the specific language governing permissions and
-#    limitations under the License.
-##############################################################################
-
-import math
-from pyquil import Program, get_qc
-from pyquil.gates import H
-from six.moves import range
-
-
-def qubits_needed(number_of_sides):
-    """
-    The number of qubits needed for a die of n faces.
-    """
-    return int(math.ceil(math.log(number_of_sides, 2)))
-
-
-def get_qvm(number_of_sides):
-    """
-    Get a QVM to simulate the requested number of sides.
-    """
-    return get_qc(f"{qubits_needed(number_of_sides)}q-qvm")
-
-
-def die_program(number_of_sides):
-    """
-    Generate a quantum program to roll a die of n faces.
-    """
-    prog = Program()
-    n_qubits = qubits_needed(number_of_sides)
-    ro = prog.declare('ro', 'BIT', n_qubits)
-    # Hadamard initialize.
-    for q in range(n_qubits):
-        prog.inst(H(q))
-    # Measure everything.
-    for q in range(n_qubits):
-        prog.measure(q, ro[q])
-    return prog
-
-
-def process_results(results):
-    """
-    Convert n digit binary result from the QVM to a value on a die.
-    """
-    raw_results = results[0]
-    processing_result = 0
-    for each_qubit_measurement in raw_results:
-        processing_result = 2*processing_result + each_qubit_measurement
-    # Convert from 0 indexed to 1 indexed
-    die_value = processing_result + 1
-    return die_value
-
-
-def roll_die(qvm, number_of_sides):
-    """
-    Roll an n-sided quantum die.
-    """
-    die_compiled = qvm.compile(die_program(number_of_sides))
-    return process_results(qvm.run(die_compiled))
-
-
-if __name__ == '__main__':
+# ----------------------------
+# Quantum Die
+# ----------------------------
+# Emily Stamm
+# Dec 30, 2018
+
+from pyquil import *
+import numpy as np
+from pyquil.quil import *
+
+def throw_polyhedral_die_helper(p,qc, num_qubits):
+	result_dict = qc.run_and_measure(p, trials = 1)
+	result = 1
+	for i in range(num_qubits):
+		result += (2**i)*result_dict[i][0] 
+	return result
+
+# throw_polyhedral_die(num_sides)
+# ----------------------------
+# return the result of throwing a num_sides sided die by running a quantum program
+
+def throw_polyhedral_die(num_sides):
+	# Number of qubits needed for computation
+	num_qubits = int(np.ceil(np.log2(num_sides)))
+	# Initialize program
+	p  = Program()
+	p.inst(H(i) for i in range(num_qubits)) 
+
+
+	qc_string = str(num_qubits) + 'q-qvm'
```

Comment:
It would be nice if the logic of handling the (pseudo-)random numbers was separate from the logic of constructing the qvm, which was separate from the logic of rejection sampling. Just my 2 cents though.